### ISSUE TITLE
vglrun doesn't wrok with some glx demos

### ISSUE THREAD

**Issue by rhjdvsgsgks** [VirtualGL/virtualgl#130]
after setup with vglserver_config , I'm trying to start glxgears with vglrun on a xserver physical connected to monitor , and it returned something like this
```
[VGL] NOTICE: Added /usr/lib to LD_LIBRARY_PATH
[VGL] Shared memory segment ID for vglconfig: 44
[VGL] VirtualGL v2.6.3 64-bit (Build 20200214)
[VGL] Opening connection to 3D X server :0
[VGL] Using Pbuffers for rendering
failed to create drawable
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  53 (X_CreatePixmap)
  Value in failed request:  0x1e
  Serial number of failed request:  31
  Current serial number in output stream:  32
```
but glxspheres doesn't have this problem . did i do something wrong that caused this ?

**Comment by dcommander** [comment]
Which O/S?  Which GPU model?  Which driver version?

**Comment by rhjdvsgsgks** [comment]
archlinux , 5700xt , mesa 20.1.4

**Comment by dcommander** [comment]
Arch is in the category of "Community Support" operating systems, meaning that "it is expected that the user community will report any incompatibilities between VirtualGL and these operating systems, as well as assist in diagnosing and fixing such incompatibilities" (https://virtualgl.org/Documentation/OSSupport).  Arch unfortunately has a tendency to break things by mis-packaging them.  I'm not sure if that's the case here, but I do know that glxgears never calls `XCreatePixmap()`.  Something is amiss.  Perhaps that something is a driver issue, or perhaps Arch packaged glxgears_pixmap from the Mesa demos rather than glxgears, or perhaps both, because VirtualGL should still work with glxgears_pixmap if the driver works properly.  Also, if you're using Arch's distribution of VirtualGL, it may be mis-packaged as well (such has happened in the past.)  You should be able to install dpkg on Arch and thus install our official .deb package.

Long story short, we don't support Arch in any official capacity, because 9 times out of 10, bugs reported with VirtualGL on Arch aren't our fault.  Also, if your Radeon RX 5700 XT is using the AMDGPU Mesa driver, there are known GLX conformance issues with that driver that prevent it from working properly with VirtualGL.

**Comment by rhjdvsgsgks** [comment]
thanks for the answer
